### PR TITLE
Make utility implement standard serializable interface

### PR DESCRIPTION
**Pull request by ghost**
Changes the serializer interface so that it provides `loads` and `dumps`. This allows the cache to either use the included smart serializer or a more simple implementation (such as just passing a reference to the `json` module) for testing.

### CODE REVIEW

**Comment by hadrien** [comment]
Really good job!

**Comment by hadrien** [comment]
Can you update changelog please?

File: pyramid_caching/serializers.py [hunk 1]
```diff
@@ -33,30 +24,62 @@ def get_serializer(config_or_request):
...
+        f = StringIO()
+        protocol = 1
+        p = pickle.Pickler(f, protocol)
+        meta = {
+            'type': adapter.name,
+            'version': SERIALIZER_META_VERSION,
+            }
+        payload = adapter.serialize(obj)
+        p.dump(meta)
+        p.dump(payload)
```
Comment (pior) [inline pyramid_caching/serializers.py:61]: Why creating 2 pickled blob? 

Is the only goal optimizing the conditions check when unpickling?

**Comment by pior** [comment]
How do we "Approve" on github ?

File: pyramid_caching/serializers.py (hunk 1 above)
Comment (ghost) [inline pyramid_caching/serializers.py:61]: @pior It's not an optimization, it's just a way to make sure that the information about the object does not conflict with the data of the state of the object. Strangely the idea comes from [ZODB](https://github.com/zopefoundation/ZODB/blob/master/src/ZODB/serialize.py).

**Pull request merged**